docs/deployment-arbitrary-cluster.md

```diff
@@ -0,0 +1,124 @@
+yum install -y NetworkManager-ovs
+systemctl daemon-reload
+systemctl restart openvswitch NetworkManager
+```
+
+On Fedora:
+
+```shell
+dnf install -y openvswitch NetworkManager-ovs
+systemctl daemon-reload
+systemctl restart openvswitch NetworkManager
+```
+
+#### OpenShift
+
+With OpenShift, the situation is a little bit difficult. If you just install
+Open vSwitch on the node, it would break openshift-sdn networking. What you can
+do instead, is to make openshift-sdn Open vSwitch instance available on the node.
+
+In order to do that, install following manifest. Please not that this method is
+not officially supported and may break your OpenShift networking.
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/openshift-ovs-vsctl.yaml
+```
+
+Once Open vSwitch is exposed to the node, install appropriate NetworkManager plugin.
+
+On CentOS:
+
+```shell
+yum install -y NetworkManager-ovs
+systemctl daemon-reload
+systemctl restart NetworkManager
+```
+
+On Fedora:
+
+```shell
+dnf install -y NetworkManager-ovs
+systemctl daemon-reload
+systemctl restart NetworkManager
+```
+
+## kubernetes-nmstate
+
+Finally we can install kubernetes-nmstate on our cluster.
+
+Please note that this will not start kubernetes-nmstate controllers on the
+cluster, only prepare it for their use. For more information, read
+[active vs. on-demand](user-guide-active-vs-on-demand.md)
+
+### Kubernetes
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/namespace.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/rbac.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/state-crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/configuration-policy-crd.yaml
+```
+
+### OpenShift
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/namespace.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/rbac.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/state-crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/configuration-policy-crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/nmstate/kubernetes-nmstate/master/manifests/examples/openshift-scc.yaml
+```
+
+You can stop here and play with the cluster on your own or continue with one of
+[user guides](user-guide.md) that will guide you through reporting of node
```

Comment:
*the other* user guides...
And this sentence also uses reporting incorrectly IMHO. Maybe use something like:
`through requesting node network states and configuring the nodes.`

Reply:
Done